### PR TITLE
[CN-545] Replace Rest client with Go for backup and refactor package structure

### PR DESCRIPTION
**Pull request by mtyazici**

### CODE REVIEW

File: controllers/hazelcast/hot_backup_controller_test.go [hunk 1]
```diff
@@ -2,151 +2,18 @@ package hazelcast
...
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	hazelcastv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
-	hzclient "github.com/hazelcast/hazelcast-platform-operator/controllers/hazelcast/client"
-	hzconfig "github.com/hazelcast/hazelcast-platform-operator/controllers/hazelcast/config"
 )
 
-func TestHotBackupReconciler_shouldSetStatusToFailedWhenHbCallFails(t *testing.T) {
```
Comment (dzeromski-hazelcast) [inline controllers/hazelcast/hot_backup_controller_test.go:25]: So we decided to drop the tests? :)

File: controllers/hazelcast/map_controller.go [hunk 2]
```diff
@@ -21,8 +21,8 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
 	hazelcastv1alpha1 "github.com/hazelcast/hazelcast-platform-operator/api/v1alpha1"
-	hzclient "github.com/hazelcast/hazelcast-platform-operator/controllers/hazelcast/client"
 	"github.com/hazelcast/hazelcast-platform-operator/internal/config"
+	hzclient "github.com/hazelcast/hazelcast-platform-operator/internal/hazelcast-client"
```
Comment (dzeromski-hazelcast) [inline controllers/hazelcast/map_controller.go:25]: nit: Can we move it to `internal/protocol/client` and not use `hazelcast-client`?

**Review by dzeromski-hazelcast — APPROVED** [review]

File: controllers/hazelcast/hot_backup_controller_test.go (hunk 1 above)
Comment (mtyazici) [reply controllers/hazelcast/hot_backup_controller_test.go:25]: We will transfer them into E2E tests in a different task

**Review by cagric0 — APPROVED** [review]
LGTM!

**Pull request merged**